### PR TITLE
✨ catalog api: new editable metadata fields

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ✨ `version_display` is now editable (i.e. can be updated in the API)
     - @odeimaiz `version_display` is _already_ part of history items (see details in [OAS](https://github.com/ITISFoundation/osparc-simcore/blob/master/services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml#L10459-L10486))
- 🎨 `history[*].compatibility.can_update_to` returns now key,version (agreed with @odeimaiz)
- ♻️ `badge` field marked as *deprecated* (agreed with @odeimaiz)
- ✨ new RPC API: `check_for_service` checks whether service exists is accessible @matusdrobuliak66 (provides a more effective call for https://github.com/ITISFoundation/osparc-simcore/pull/6085)
- ♻️  Removes `DEV_FEATURE` constraint in `/dev` entrypoints  to permit https://github.com/ITISFoundation/osparc-simcore/pull/6016 to run in staging+production (FYI @odeimaiz)

## Related issue/s

- Part of #5964
- Implements a call for https://github.com/ITISFoundation/osparc-simcore/pull/6085

## How to test

catalog test

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by odeimaiz — COMMENTED** [review]
👍

File: api/specs/web-server/_projects_groups.py [hunk 1]
```diff
@@ -25,53 +24,40 @@
...
-assert_handler_signature_against_model(list_project_groups, ProjectPathParams)
-
-
 @router.put(
     "/projects/{project_id}/groups/{group_id}",
     response_model=Envelope[ProjectGroupGet],
 )
-async def update_project_group(
-    project_id: ProjectID, group_id: GroupID, body: _ProjectsGroupsBodyParams
+async def replace_project_group(
```
Comment (matusdrobuliak66) [inline api/specs/web-server/_projects_groups.py:51]: Thanks for unification :)

Comment (pcrespov) [reply api/specs/web-server/_projects_groups.py:51]: @matusdrobuliak66 a test was failing

File: packages/models-library/src/models_library/api_schemas_catalog/services.py [hunk 2]
```diff
@@ -263,23 +263,23 @@ class Config:
                         {
                             "version": "2.0.0",
                             "compatibility": {
-                                "canUpdateTo": _EXAMPLE_SLEEPER["version"],
+                                "canUpdateTo": {"version": _EXAMPLE_SLEEPER["version"]},
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_catalog/services.py:266]: Q: how do you allow to update to a different service key?

File: packages/models-library/src/models_library/services_base.py [hunk 3]
```diff
@@ -20,7 +20,7 @@ class Config:
         frozen = True
 
 
-class ServiceBase(BaseModel):
+class ServiceDisplay(BaseModel):
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_base.py:23]: ServiceBaseInfo? Display in this context of a backend service is a bit out of context no?

File: packages/models-library/src/models_library/services_history.py [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
-from models_library.services_base import ServiceKeyVersion
 from pydantic import BaseModel, Field
 
-from .services_types import ServiceVersion
+from .services_types import ServiceKey, ServiceVersion
 from .utils.change_case import snake_to_camel
 
 
+class CompatibleService(BaseModel):
+    key: ServiceKey | None = Field(
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_history.py:11]: ok

File: packages/models-library/src/models_library/api_schemas_catalog/services.py (hunk 2 above)
Comment (sanderegg) [reply packages/models-library/src/models_library/api_schemas_catalog/services.py:266]: ok found it. but you might want an example here.

File: packages/models-library/src/models_library/services_history.py [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
+        description="If None, it refer to current service. Used only for inter-service compatibility",
+    )
+    version: ServiceVersion
+
+
+assert set(CompatibleService.__fields__) == set(CompatibleService.__fields__)  # nosec
+
+
 class Compatibility(BaseModel):
     # NOTE: as an object it is more maintainable than a list
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_history.py:22]: not sure the note makes sense anymore?

File: packages/models-library/src/models_library/services_metadata_editable.py [hunk 6]
```diff
@@ -15,18 +15,21 @@
...
     description: str | None
+    version_display: str | None = None
+
+    # Below fields only in the database ----
     deprecated: datetime | None = Field(
         default=None,
-        description="If filled with a date, then the service is to be deprecated at that date (e.g. cannot start anymore)",
+        description="Owner can set the date to retire the service. Three possibilities:"
+        "If None, the service is marked as `published`;"
+        "If now<deprecated the service is marked as dprecated;"
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_metadata_editable.py:30]:
```suggestion
        "If now<deprecated the service is marked as deprecated;"
```

File: packages/models-library/src/models_library/services_history.py [hunk 7]
```diff
@@ -56,7 +64,12 @@ class Config:
                     "version_display": "Matterhorn",
                     "released": "2024-06-20T18:49:17",
                     "retired": "2034-06-20T00:00:00",
-                    "compatibility": {"can_update_to": "0.9.10"},
+                    "compatibility": {
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_history.py:67]: ah so here you have an example

File: services/catalog/src/simcore_service_catalog/db/repositories/services.py [hunk 8]
```diff
@@ -271,6 +272,28 @@ async def update_service(
 
     # NEW CRUD on services ------
```
Comment (sanderegg) [inline services/catalog/src/simcore_service_catalog/db/repositories/services.py:273]: maybe remove that comment

File: services/catalog/tests/unit/test_services_compatibility.py [hunk 9]
```diff
@@ -279,5 +279,5 @@ async def test_evaluate_service_compatibility_map_with_deprecated_versions(
         compatibility_map[ServiceVersion("1.0.0")] is None
     )  # cannot upgrade to deprecated 1.0.1
     assert compatibility_map[ServiceVersion("1.0.1")] is None  # Deprecated version
-    assert compatibility_map[ServiceVersion("1.2.0")].can_update_to == "1.2.5"
+    assert compatibility_map[ServiceVersion("1.2.0")].can_update_to.version == "1.2.5"
```
Comment (sanderegg) [inline services/catalog/tests/unit/test_services_compatibility.py:282]: Q: where do you test updating to a different key?

File: services/web/server/src/simcore_service_webserver/catalog/_handlers.py [hunk 10]
```diff
@@ -142,7 +139,6 @@ async def dev_get_service(request: Request):
     f"{VTAG_DEV}/catalog/services/{{service_key}}/{{service_version}}",
     name="dev_update_service",
 )
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/catalog/_handlers.py:141]: Q: when do you plan to remove the /dev api entrypoint?

**Review by sanderegg — APPROVED** [review]
thanks! looking forward to seeing the whole thing in motion.

File: services/catalog/tests/unit/test_services_compatibility.py (hunk 9 above)
Comment (GitHK) [reply services/catalog/tests/unit/test_services_compatibility.py:282]: I second the question

**Review by GitHK — APPROVED** [review]

File: services/catalog/src/simcore_service_catalog/db/repositories/_services_sql.py [hunk 11]
```diff
@@ -264,6 +267,37 @@ def list_latest_services_with_history_stmt(
...
+        .where(
+            (services_meta_data.c.key == service_key)
+            & (services_meta_data.c.version == service_version)
+            & (user_to_groups.c.uid == user_id)
+            & (services_access_rights.c.product_name == product_name)
+            & access_rights
+        )
+        .limit(1)
+    )
+    return sa.select(sa.exists(subquery))
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/catalog/src/simcore_service_catalog/db/repositories/_services_sql.py:298]: https://docs.sqlalchemy.org/en/20/tutorial/data_select.html#exists-subqueries

File: packages/models-library/src/models_library/services_history.py (hunk 5 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/services_history.py:22]: haha this is obviously a mistake :-) thx

File: services/catalog/src/simcore_service_catalog/db/repositories/services.py (hunk 8 above)
Comment (pcrespov) [reply services/catalog/src/simcore_service_catalog/db/repositories/services.py:273]: ok

File: services/catalog/tests/unit/test_services_compatibility.py (hunk 9 above)
Comment (pcrespov) [reply services/catalog/tests/unit/test_services_compatibility.py:282]: @sanderegg @GitHK  see `async def test_evaluate_service_compatibility_map_with_other_service` in the same file

File: services/web/server/src/simcore_service_webserver/catalog/_handlers.py (hunk 10 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/catalog/_handlers.py:141]: As soon as the front-end is in, we will make a PR including the change in the front-end and backend

**Pull request merged**